dev/opentitan.nix

```diff
@@ -6,7 +6,7 @@
   ncurses5-fhs,
```

Comment:
Re. ecea89554, the commit message doesn't seem to match the changes again? This appears to just be a snake -> kebab change for the nix binding of opentitan's python environment.
I think the spirit of the message is correct though, we should probably give the naming another though at some time. Ideally 'python' is distinct from a 'pythonEnv', but also should the pname contain the project that particular env is intended for? I'm undecided. I almost think that it would be better if we could namespace as e.g. `packages.${system}.opentitan.pythonEnv`, but flakes doesn't allow that output schema.